BugsnagPerformance/Assets/BugsnagPerformance/Scripts/Models/AttributeValueModel.cs

```diff
@@ -43,4 +45,85 @@ public AttributeDoubleValueModel(double theValue)
             doubleValue = theValue;
         }
     }
+
+    public class AttributeStringArrayValueModel : AttributeValueModel
+    {
+        public ArrayValueModel arrayValue;
+
+        public AttributeStringArrayValueModel(string[] values)
+        {
+            arrayValue = new ArrayValueModel(values);
+        }
+    }
+
+    public class AttributeIntArrayValueModel : AttributeValueModel
+    {
+        public ArrayValueModel arrayValue;
+
+        public AttributeIntArrayValueModel(int[] values)
+        {
+            arrayValue = new ArrayValueModel(values);
+        }
+    }
+
+    public class AttributeBoolArrayValueModel : AttributeValueModel
+    {
+        public ArrayValueModel arrayValue;
+
+        public AttributeBoolArrayValueModel(bool[] values)
+        {
+            arrayValue = new ArrayValueModel(values);
+        }
+    }
+
+    public class AttributeDoubleArrayValueModel : AttributeValueModel
+    {
+        public ArrayValueModel arrayValue;
+
+        public AttributeDoubleArrayValueModel(double[] values)
+        {
+            arrayValue = new ArrayValueModel(values);
+        }
+    }
+
+    public class ArrayValueModel
+    {
+        public List<AttributeValueModel> values;
+
+        public ArrayValueModel(string[] values)
+        {
+            this.values = new List<AttributeValueModel>();
+            foreach (var value in values)
+            {
+                this.values.Add(new AttributeStringValueModel(value));
+            }
+        }
```

Comment:
Could this work maybe be moved to when we create the payloads instead of when the attributes are set? I think it would be a bit surprising to have `Span.SetAttribute` be an O(n) operation for most customers.

Reply:
Done 👍